pkg/templates/nstemplatetier_generate_test.go

```diff
@@ -0,0 +1,103 @@
+package templates_test
+
+import (
+	"testing"
+
+	"github.com/codeready-toolchain/host-operator/pkg/templates"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestNSTemplateTierGenerator(t *testing.T) {
+
+	t.Run("basic", func(t *testing.T) {
+		// given
+		tier := "basic"
+		// when
+		result, err := templates.GenerateNSTemplateTierManifest(tier)
+		// then
+		require.NoError(t, err)
+		expected := `kind: NSTemplateTier
```

Comment:
So, every time we change any of the templates we will need to update this expected string too. It can be pretty inconvenient. If we use some util to re-generate the template then we may compromise the test.

What if we:
- construct NSTemplateTier structs (https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go#L31) for expected tiers (all of them) with all not-optional fields set up.
- unmarshal all the generated tier yaml manifests into NSTemplateTier structs
- smart assert equality of these expected and actual tiers.

Expected revisions and template values could be loaded from Asset.

Regarding revision values. The revisions (commit SHAs) will be actually different. They are auto-generated in build time. So they will be different in the binaries built from master and from the PR branch. So, we can't hardcode them in the tests. Unless I'm missing something.
We will need to load them from Asset and check if they match.

Also, in our API (https://github.com/codeready-toolchain/api/blob/master///pkg/apis/toolchain/v1alpha1/nstemplateset_types.go#L33) templates are currently optional strings and are supposed to be used for custom templates only. But it seems that you actually are going to pass the templates in the tier manifest as yaml and it's not optional (so it's not for custom templates anymore).
We need to align our API with that approach.
We will probably need to change the Template field of the Namespace struct from `string` to https://github.com/openshift/api/blob/master/template/v1/types.go#L15 and make it required.

Reply:
@alexeykazakov I'm going to see how to address your concerns with regards to tests stability as the templates and commit hash evolve with time.